types/seeed/senseCAPS2103/uplink.js

```diff
@@ -0,0 +1,91 @@
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const data = {};
+  let pointer = 0;
+
+  while (pointer < payload.length) {
+    const measurementID = Hex.hexLittleEndianToBigEndian(
+      payload.substr((pointer += 2), 4),
+      false,
+    );
+    switch (measurementID) {
+      case 7: {
+        const batteryLevel = Hex.hexLittleEndianToBigEndian(
+          payload.substr((pointer += 4), 4),
+          false,
+        );
+        const sendInterval = Hex.hexLittleEndianToBigEndian(
+          payload.substr((pointer += 4), 4),
+          false,
+        );
+        pointer += 4;
+        emit("sample", {
+          data: { batteryLevel, sendInterval },
+          topic: "lifecycle",
+        });
+        break;
+      }
+      case 4097:
+        data.temperature =
+          Hex.hexLittleEndianToBigEndian(
```

Comment:
please split the calculation into multiple lines (for all cases), it is not readable like this